### PR TITLE
Upload Compute@Edge packages

### PR DESCRIPTION
**Pull request by marsavar**
## What does this change?

This PR adds a new method to the API client that allows package uploads to Fastly.

## Why?

This is part of the plan to make Compute@Edge packages deployable by Riff-Raff at the Guardian.
Conveniently, the Fastly API has a PUT endpoint that allows package uploads. Specifically, a package is a `tar.gz` file containing a Webassembly binary. It is generated when using the Fastly CLI and running the command `fastly compute build`.

This library currently does not support `multi-part` content types so I've had to change the `execute` method (which in turn calls the private `build` method) to support a new optional parameter. I've made sure this defaults to `None` in order not to introduce any breaking changes.

I've also added a test for this method, making sure the test is skipped if the Compute@Edge service path and version are missing from the config file, as I think it is unreasonable to expect people testing this library to download the Fastly CLI and create an Edge service just for tests to pass.

## How to test

1) Add `packagePath` and `packageVersionId` to your `~/.config/fastly/fastlyApiClientTest` file
2) `test` from the `sbt` shell

```
[info] FastlyApiClientTest:
[info] Feature: stats
[info]   Scenario: stats
[info]   Scenario: stats with field filter
[info]   Scenario: aggregate
[info]   Scenario: service
[info]   Scenario: service with field filter
[info]   Scenario: usage
[info]   Scenario: usage by grouped by service
[info]   Scenario: regions
[info]   Scenario: stats region list
[info] Feature: Service
[info]   Scenario: list
[info] Feature: Package
[info]   Scenario: upload package
[info] Run completed in 3 seconds, 521 milliseconds.
[info] Total number of tests run: 11
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 11, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 4 s, completed 27 Apr 2023, 17:34:17
```

Additionally, I have checked in the Fastly interface that the package upload was successful.

image

### CODE REVIEW

File: src/test/scala/com/gu/fastly/api/FastlyApiClientTest.scala [hunk 1]
```diff
@@ -107,14 +107,32 @@ class FastlyApiClientTest extends AnyFeatureSpec with Matchers {
     }
   }
 
-  Feature("Servcie") {
+  Feature("Service") {
```
Comment (AshCorr) [inline src/test/scala/com/gu/fastly/api/FastlyApiClientTest.scala:110]: thank you for your servcie in cleaning up our code!

**Review by AshCorr — APPROVED** [review]

**Comment by AshCorr** [comment]
Awesome 👍

File: src/main/scala/com/gu/fastly/api/FastlyApiClient.scala [hunk 2]
```diff
@@ -52,24 +52,25 @@ case class FastlyApiClient(apiKey: String, serviceId: String, config: Option[Asy
...
-   *                    This is the package that is generated by running the Fastly CLI with
-   *                    the command `fastly compute build`. The package is created in the `pkg/` folder.
+   * @param package The Compute@Edge package to upload to the Fastly API endpoint.
+   *                This is a `.tar.gz` file that is generated by running the Fastly CLI with
+   *                the command `fastly compute build`. The package is created in the `pkg/` folder
+   *                within your Compute@Edge project.
    * @return
    */
-  def packageUpload(serviceId: String, versionId: Int, packagePath: String): Future[Response] = {
+  def packageUpload(serviceId: String, versionId: Int, `package`: File): Future[Response] = {
```
Comment (AshCorr) [inline src/main/scala/com/gu/fastly/api/FastlyApiClient.scala:66]: Alternatively have the consumer pass in an array of bytes!
